### PR TITLE
Base Curve and Surface objects

### PR DESCRIPTION
**Pull request by tomvanmele**
### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [x] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by tomvanmele** [comment]
added a bunch of files that can be used to test in Rhino and Blender (if `compas_occ` is installed). they should become part of the tutorial at some point...

**Comment by tomvanmele** [comment]
will also add dedicated artists for curves and surfaces, but in a next PR...

**Comment by tomvanmele** [comment]
functionality is fully compatible and tested with `compas_occ`

**Comment by tomvanmele** [comment]
the general curve and surface will be useful for BRep geometries (noticed the need for general curve and surface objects while working on the BRep stuff of `compas_occ`)

**Comment by tomvanmele** [comment]
@gonzalocasas this one is now ready

**Comment by tomvanmele** [comment]
@gonzalocasas @Licini ping

**Comment by tomvanmele** [comment]
it looks a bit daunting but you can just ignore the files in the tutorial folder. they are just for testing and i will add them to a proper tutorial at some point...

**Review by Licini — APPROVED** [review]
LGTM!

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +43,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Moved functionality of `compas.utilities.coercion` to `compas.data`.
 * Fixed bug in `compas.geometry.NurbsSurface.to_triangles()`.
 * Renamed docs site folders `latest` to `stable` and `dev` to `latest`.
+* Rebased `compas.geometry.NurbsCurve` on `compas.geometry.Curve`.
```
Comment (gonzalocasas) [inline CHANGELOG.md:46]: What means `rebased` in this context? Changed the base class?

File: src/compas/artists/surfaceartist.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+
+class SurfaceArtist(Artist):
+    """Base class for artists for surfaces.
+
+    Parameters
+    ----------
+    surface: :class:`~compas.geometry.Surface`
+        The surface geometry.
+    color : tuple[float, float, float] | :class:`~compas.colors.Color`, optional
```
Comment (gonzalocasas) [inline src/compas/artists/surfaceartist.py:16]: the pipe is a new thing for `or`, I think we've been using the word `or` or the `Union[]` type so far

File: src/compas_rhino/geometry/surfaces/nurbs.py [hunk 3]
```diff
@@ -9,14 +9,46 @@
...
-from compas_rhino.conversions import plane_to_compas_frame
-from compas_rhino.conversions import box_to_compas
-from compas_rhino.conversions import RhinoCurve
+
+from .surface import RhinoSurface
 
 import Rhino.Geometry
 
 
+class Points:
```
Comment (gonzalocasas) [inline src/compas_rhino/geometry/surfaces/nurbs.py]: is this missing the `(object)` part?

Comment (gonzalocasas) [reply src/compas_rhino/geometry/surfaces/nurbs.py]: and.. should it be called something more verbose, like `SurfacePoints`?

**Review by gonzalocasas — APPROVED** [review]
LGTM!

File: CHANGELOG.md (hunk 1 above)
Comment (tomvanmele) [reply CHANGELOG.md:46]: yes

File: src/compas/artists/surfaceartist.py (hunk 2 above)
Comment (tomvanmele) [reply src/compas/artists/surfaceartist.py:16]: i know but it is the preferred over `Union` in the latest versions of the python docs

Comment (gonzalocasas) [reply src/compas/artists/surfaceartist.py:16]: ah ok! good, makes sense to me, I never liked `Union` much.. so, from now, `|` everywhere, right?

Comment (tomvanmele) [reply src/compas/artists/surfaceartist.py:16]: yes

**Pull request merged**